### PR TITLE
Ms3 connect api

### PR DESCRIPTION
**Pull request by Bishoy-Samwel**
- [x] Read the[ Bookstore API ](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739)documentation to learn how to use the API. 
- [x] Refactor your add book and remove book features to persist your changes in the server.
- [x] Add book and remove book should work in the same way after the refactor.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,32 +1,128 @@
...
+
 // Actions Types
 const ADD_BOOK = 'bookStore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
+const LOAD_BOOKS_IN_PROGRESS = 'LOAD_BOOKS_IN_PROGRESS';
+const LOAD_BOOKS_SUCCESS = 'LOAD_BOOKS_SUCCESS';
+const LOAD_BOOKS_FAILURE = 'LOAD_BOOKS_FAILURE';
 
-// Reducer
+// Reducers
```
Comment (afanimartin) [inline src/redux/books/books.js:11]: Please remove this comment and any other comments in the codebase. Comments make our codebase look ugly.

File: src/components/Book.js [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
-      <button type="button" onClick={() => dispatch(removeBook(book.id))}>Remove</button>
+      <button type="button" onClick={() => dispatch(removeBookRequest(book.item_id))}>Remove</button>
       <button type="button">Edit</button>
     </li>
   );
 };
 
 Book.propTypes = {
-  book: PropTypes.shape.isRequired,
+  // eslint-disable-next-line react/forbid-prop-types
```
Comment (afanimartin) [inline src/components/Book.js:22]: Do not disable eslint rules. Please fix them instead.

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @Bishoy-Samwel, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (Bishoy-Samwel) [reply src/redux/books/books.js:11]: Comments are very useful to illustrate the job of each code block. 

> These are called Clarification comments
> Clarification comments are intended for anyone (including your future self) who may need to maintain, refactor, or extend your code.

 I suggest having a look here https://www.freecodecamp.org/news/code-comments-the-good-the-bad-and-the-ugly-be9cc65fbf83/

File: src/components/Book.js (hunk 2 above)
Comment (Bishoy-Samwel) [reply src/components/Book.js:22]: Sure I will do that thanks for the review 👍🏼

**Pull request merged**